From 2fa2426d7c878c3b1d0be41fee387d8227646bb4 Mon Sep 17 00:00:00 2001
From: InterLinked1 <24227567+InterLinked1@users.noreply.github.com>
Date: Tue, 29 Jul 2025 17:42:03 -0400
Subject: [PATCH 4/6] dahdi_dummy: Use module_init/module_exit instead of
 init_module/cleanup_module.

Since kernel commit 4fab2d7628dd38f3fa8a5e615199350ecaeb17a8, building
dahdi_dummy fails with an objtool failure since init_module/cleanup_module
are deprecated. Update dahdi_dummy to use module_init and module_exit, just
like all other modules.

Resolves: #94
---
 drivers/dahdi/dahdi_dummy.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

--- a/drivers/dahdi/dahdi_dummy.c
+++ b/drivers/dahdi/dahdi_dummy.c
@@ -209,7 +209,7 @@ static int dahdi_dummy_initialize(struct
 	return res;
 }
 
-int init_module(void)
+static int __init dummy_init(void)
 {
 	int res;
 	ztd = kzalloc(sizeof(*ztd), GFP_KERNEL);
@@ -252,8 +252,7 @@ int init_module(void)
 	return 0;
 }
 
-
-void cleanup_module(void)
+static void __exit dummy_exit(void)
 {
 #if defined(USE_HIGHRESTIMER)
 	/* Stop high resolution timer */
@@ -274,3 +273,6 @@ module_param(debug, int, 0600);
 MODULE_DESCRIPTION("Timing-Only Driver");
 MODULE_AUTHOR("Robert Pleh <robert.pleh@hermes.si>");
 MODULE_LICENSE("GPL v2");
+
+module_init(dummy_init);
+module_exit(dummy_exit);
